### ISSUE TITLE
plugin @id:ms-vscode.csharp was not found

### ISSUE THREAD

**Issue by zhaolihang** [OmniSharp/omnisharp-vscode#3624]
Issue Type: <b>Bug</b>

I use "Debugger for Unity" , but it depends "ms-vscode.csharp", and "ms-vscode.csharp" was not found.

Extension version: 1.21.13
VS Code version: Code 1.42.1 (c47d83b293181d9be64f27ff093689e8e7aed054, 2020-02-11T14:45:59.656Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9700KF CPU @ 3.60GHz (8 x 3600)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.92GB (10.80GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by lindsaytalbot** [comment]
Also getting this issue. Also get the "The C# extension is recommended for this file type" even though the extension is installed. Intellisense is broken. References are broken. Just a text editor now

**Comment by lindsaytalbot** [comment]
Rolling back to 1.21.12 resolved the issue
https://github.com/OmniSharp/omnisharp-vscode/issues/3623

**Comment by fdinardo** [comment]
look like the new id of the extension is `ms-dotnettools.csharp` instead of `ms-vscode.csharp`.

@lindsaytalbot if I try to rollback the latest c# extension I get the error: Unable to find the extenstion version.

The suggested action is to download it manually and install it. 

I think there are some extensions that uses the old id as dependency.

**Comment by roxberry** [comment]
VS Code on Mac:

Version: 1.42.1
Commit: c47d83b293181d9be64f27ff093689e8e7aed054
Date: 2020-02-11T14:44:27.652Z
Electron: 6.1.6
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 19.4.0

On open of a project and CS file:

Prompt to install C#:
"The 'C#' extension is recommended for this file type."

Click Install, Extensions: Marketplace opens:
@id:ms-vscode.csharp
"No extensions found."

Additional error message:
"Cannot activate the 'Roslynator' extension because it depends on an unknown 'ms-vscode.csharp' extension ."

**Comment by JasonRTBond** [comment]
On a similar environment to roxberry, I have the same issue.

Work around: install 1.21.12 manually via https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ms-dotnettools/vsextensions/csharp/1.21.12/vspackage

**Comment by warbands** [comment]
Any easy way to roll back to 1.21.12 on the Mac?

**Comment by warbands** [comment]
Trying to use the Editor Extension Settings icon to install a different version from within VSCode fails. .vsix files don't seem to be recognized on macOS

**Comment by JasonRTBond** [comment]
@warbands From the extensions panel you can click the three little dots and find the option to install from a .vsix. Annoying that the extension doesn't just open with Code.

Aside problem: I chose the older version via Code's "install other version..." option and selected to manually install it. It then opened a link to the VSIX for the **current** version. I had to manually edit the URL to get 1.21.12.

**Comment by diegoperes** [comment]
I have the same problem, it seems that it is not identifying the installed extension ms-dotnettools.csharp and is suggesting the ms-vscode.csharp that does not exist.

**Comment by njelly** [comment]
Any solution for macOS? .vsix seems to be windows only

**Comment by JasonRTBond** [comment]
> Any solution for macOS? .vsix seems to be windows only

I’m on macOS. See notes above. In short, the vsix extension doesn’t open automatically but you can still open it.

**Comment by njelly** [comment]
Oh!!! The "three little dots". Thank you! I no longer get the error and I will be warier of updates from now on.

**Comment by cartermp** [comment]
Thanks for reporting. The issue is due to the publisher rename, which has broken the unity extension. There is a pull request out on the unity extension, and once that is merged and updates go through, it should be resolved. Closing as a duplicate of https://github.com/OmniSharp/omnisharp-vscode/issues/3620
